src/plugin/ipc/file/fileconnlist.cpp

```diff
@@ -490,13 +490,22 @@ FileConnList::recreateShmFileAndMap(const ProcMapsArea &area)
     JASSERT(fd != -1) (JASSERT_ERRNO);
   }
 
-  // Get to the correct offset.
-  JASSERT(lseek(fd, area.offset, SEEK_SET) == area.offset) (JASSERT_ERRNO);
-
-  // Now populate file contents from memory.
-  JASSERT(Util::writeAll(fd, area.addr, area.size) == (ssize_t)area.size)
-    (JASSERT_ERRNO);
+  // Unlink (the area was originally unlinked)
+  unlink(area.name);
+
+  // Restore SHM area mapping
+  // The mmap in restoreShmArea will clear the content in the area because the
+  // file has been unlinked, so we have to save the content first into a
+  // temp buffer
```

Comment:
If I remember correctly, this bug appeared because area.name was originally a file based on hugepages.  Is this bug fix specific to a bug that appeared with a memory segment of hugepages,  Or does the bug appear with an ordinary mmap using MAP_SHARED also?

If it's specific to hugepages, we need to add something about that in the comment.  If it's about an ordinary mmap with MAP_SHARED, can we further simplify the example that led to this bug?

Reply:
This does not seem to be related to the hugepages at all.